From a3b1e3af333c34d19e288cd77a208440f0fe6820 Mon Sep 17 00:00:00 2001
From: Michael Webster <miketwebster@gmail.com>
Date: Sun, 30 Jun 2019 18:44:10 -0400
Subject: [PATCH] csd-power-manager.c: Define UP_DEVICE_LEVEL_NONE for
 ourselves if it's missing.  UpDeviceLevel was new in 0.99.0.

related commit in cinnamon: linuxmint/cinnamon@4d1da807ee6e73

Fixes #258
---
 plugins/power/csd-power-manager.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugins/power/csd-power-manager.c b/plugins/power/csd-power-manager.c
index 9ca0852..fae6510 100644
--- a/plugins/power/csd-power-manager.c
+++ b/plugins/power/csd-power-manager.c
@@ -4414,6 +4414,10 @@ csd_power_manager_finalize (GObject *object)
         G_OBJECT_CLASS (csd_power_manager_parent_class)->finalize (object);
 }
 
+#if !UP_CHECK_VERSION(0,99,0)
+#define UP_DEVICE_LEVEL_NONE 1
+#endif
+
 static GVariant *
 device_to_variant_blob (UpDevice *device)
 {
